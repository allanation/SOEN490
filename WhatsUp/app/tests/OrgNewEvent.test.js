import "react-native";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from "@testing-library/react-native";
import OrgNewEventScreen from "../screens/OrgNewEvent";
import OrgPOCScreen from "../screens/OrgPointOfContact";
import { Alert } from "react-native";

jest.useFakeTimers();

// it("Renders Create New Event First Page Correctly", () => {
//   const tree = render(
//     <NavigationContainer>
//       <OrgNewEventScreen />
//     </NavigationContainer>
//   ).toJSON();
//   expect(tree).toMatchSnapshot();
// });

test("If the Event Title is missing, an alert should be prompted", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  jest.spyOn(Alert, "alert");

  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Event Title"), "")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Organization Name"), "OrganizationTest")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Location"), "Montreal")
  );
  await waitFor(() =>
  fireEvent.changeText(screen.getByPlaceholderText("Description"), "Description Test")
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
    expect(Alert.alert);
  });
});

test("If the Organizer Name is missing, an alert should be prompted", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  jest.spyOn(Alert, "alert");

  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Event Title"), "Title Test")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Organization Name"), "")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Location"), "Montreal")
  );
  await waitFor(() =>
  fireEvent.changeText(screen.getByPlaceholderText("Description"), "Description Test")
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
    expect(Alert.alert);
  });
});

test("If the Location is missing, an alert should be prompted", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  jest.spyOn(Alert, "alert");

  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Event Title"), "Title Test")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Organization Name"), "OrganizationTest")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Location"), "")
  );
  await waitFor(() =>
  fireEvent.changeText(screen.getByPlaceholderText("Description"), "Description Test")
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
    expect(Alert.alert);
  });
});

test("If the Description is missing, an alert should be prompted", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  jest.spyOn(Alert, "alert");

  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Event Title"), "Title Test")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Organization Name"), "OrganizationTest")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Location"), "Montreal")
  );
  await waitFor(() =>
  fireEvent.changeText(screen.getByPlaceholderText("Description"), "")
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
    expect(Alert.alert);
  });
});

test("If the Cover Image is missing, an alert should be prompted", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  jest.spyOn(Alert, "alert");

  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Event Title"), "Title Test")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Organization Name"), "OrganizationTest")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Location"), "Montreal")
  );
  await waitFor(() =>
  fireEvent.changeText(screen.getByPlaceholderText("Description"), "Description Test")
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
    expect(Alert.alert);
  });
});

test("Successfully choosing studying image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("studyingImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully choosing McGill image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("mcGillImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully choosing park image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("parkImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully choosing Concordia image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("concordiaImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully choosing graduation image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("graduationImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully choosing auditorium image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("auditoriumImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully choosing frosh image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("froshImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully choosing art image as cover image", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("artImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

});

test("Successfully go to next page when all the mandatory fields are filled", async () => {
  render(
    <NavigationContainer>
      <OrgNewEventScreen />
    </NavigationContainer>
  );

  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Event Title"), "Title Test")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Organization Name"), "Organization Test")
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Location"), "Montreal")
  );
  await waitFor(() =>
  fireEvent.changeText(screen.getByPlaceholderText("Description"), "Description Test")
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("chooseImage"));
  });
  
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("sportsImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitImage"));
  });

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
  });
  render(
    <NavigationContainer>
      <OrgPOCScreen />
    </NavigationContainer>
  );
  expect(screen.getByPlaceholderText("Name")).toBeTruthy();
});