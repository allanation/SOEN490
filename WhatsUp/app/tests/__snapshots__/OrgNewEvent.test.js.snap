// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Create New Event First Page Correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "marginTop": 30,
        "padding": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "elevation": 2,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "marginLeft": 2,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Helvetica Neue",
            "marginLeft": "1%",
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Create New Event
    </Text>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Helvetica Neue",
            "marginLeft": "1%",
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "color": "#747474",
            "fontSize": 20,
            "marginBottom": 6,
            "marginLeft": "1%",
            "paddingVertical": 6,
          },
        ]
      }
    >
      Please fill the following information
    </Text>
  </View>
  <RCTScrollView
    style={
      Object {
        "paddingTop": 20,
      }
    }
  >
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#989797",
                "borderRadius": 12,
                "borderWidth": 1,
                "marginLeft": "1%",
                "marginVertical": 10,
                "opacity": 0.9,
                "paddingHorizontal": 12,
                "paddingVertical": 7,
                "width": "98%",
              },
              Object {
                "elevation": 4,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              },
            ]
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Event Title"
            placeholderTextColor="grey"
            style={
              Object {
                "color": "black",
                "fontFamily": "Helvetica Neue",
                "fontSize": 18,
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#989797",
                "borderRadius": 12,
                "borderWidth": 1,
                "marginLeft": "1%",
                "marginVertical": 10,
                "opacity": 0.9,
                "paddingHorizontal": 12,
                "paddingVertical": 7,
                "width": "98%",
              },
              Object {
                "elevation": 4,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              },
            ]
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Organization Name"
            placeholderTextColor="grey"
            style={
              Object {
                "color": "black",
                "fontFamily": "Helvetica Neue",
                "fontSize": 18,
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#989797",
                "borderRadius": 12,
                "borderWidth": 1,
                "marginLeft": "1%",
                "marginVertical": 10,
                "opacity": 0.9,
                "paddingHorizontal": 12,
                "paddingVertical": 7,
                "width": "98%",
              },
              Object {
                "elevation": 4,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              },
            ]
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Location"
            placeholderTextColor="grey"
            style={
              Object {
                "color": "black",
                "fontFamily": "Helvetica Neue",
                "fontSize": 18,
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#989797",
                "borderRadius": 12,
                "borderWidth": 1,
                "marginLeft": "1%",
                "marginVertical": 10,
                "opacity": 0.9,
                "paddingHorizontal": 12,
                "paddingVertical": 7,
                "width": "98%",
              },
              Object {
                "elevation": 4,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              },
            ]
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Link for ticket purchase (optional)"
            placeholderTextColor="grey"
            style={
              Object {
                "color": "black",
                "fontFamily": "Helvetica Neue",
                "fontSize": 18,
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#989797",
                "borderRadius": 12,
                "borderWidth": 1,
                "marginLeft": "1%",
                "marginVertical": 10,
                "opacity": 0.9,
                "paddingHorizontal": 12,
                "paddingVertical": 7,
                "width": "98%",
              },
              Object {
                "color": "#989797",
                "elevation": 4,
                "height": 100,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
                "textAlignVertical": "top",
              },
            ]
          }
        >
          <TextInput
            multiline={true}
            numberOfLines={5}
            onChangeText={[Function]}
            placeholder="Description"
            placeholderTextColor="grey"
            style={
              Object {
                "color": "black",
                "fontFamily": "Helvetica Neue",
                "fontSize": 18,
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "marginVertical": 10,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#989797",
                "fontSize": 18,
                "width": "90%",
              }
            }
          >
            Add Cover Image
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#00c0a4",
          "borderRadius": 20,
          "elevation": 4,
          "marginLeft": "1%",
          "marginVertical": 40,
          "opacity": 1,
          "padding": 12,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 3,
          "width": "90%",
        }
      }
      testID="nextButton"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Helvetica Neue",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            undefined,
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</View>
`;
